### PR TITLE
Bug 1813741: [ovirt] allow template parameters customization (mem/cpu/disk) in the machine

### PR DESCRIPTION
**Pull request by rgolangh**
Update the provider spec with mem/cpu/disk properties

### CODE REVIEW

**Comment by rgolangh** [comment]
/hold 
Still needs verification

File: pkg/apis/ovirtprovider/v1beta1/types.go [hunk 1]
```diff
@@ -49,7 +49,23 @@ type OvirtMachineProviderSpec struct {
 	// the oVirt cluster this VM instance belongs too
 	ClusterId string `json:"cluster_id"`
 
+	// Sockets is the number of sockets for a VM.
+	// Total CPUs is Sockets * Cores * Threads
```
Comment (bennyz) [inline pkg/apis/ovirtprovider/v1beta1/types.go]: Maybe surround with brackets (Sockets * Cores * Threads) to make it clearer it's a calculation

File: pkg/apis/ovirtprovider/v1beta1/types.go [hunk 2]
```diff
@@ -49,7 +49,23 @@ type OvirtMachineProviderSpec struct {
...
+	Sockets int32 `json:"sockets"`
 
+	// Cores is the number of cores per socket.
+	// Total CPUs is Sockets * Cores * Threads
+	Cores int32 `json:"cores"`
+
+	// Thread is the number of thread per core.
```
Comment (bennyz) [inline pkg/apis/ovirtprovider/v1beta1/types.go]: threads maybe?

File: pkg/apis/ovirtprovider/v1beta1/types.go (hunk 1 above)
Comment (rgolangh) [reply pkg/apis/ovirtprovider/v1beta1/types.go]: ack

File: pkg/apis/ovirtprovider/v1beta1/types.go (hunk 2 above)
Comment (rgolangh) [reply pkg/apis/ovirtprovider/v1beta1/types.go]: ack

File: pkg/cloud/ovirt/clients/machineservice.go [hunk 3]
```diff
@@ -160,6 +198,50 @@ func (is *InstanceService) InstanceCreate(
...
+		if disk.MustBootable() {
+			// found the os disk
+			bootableDiskAttachment = disk
+		}
+	}
+	if bootableDiskAttachment == nil {
+		return fmt.Errorf("the VM %s(%s) doesn't have a bootable disk - was Blank template used by mistake?",
+			createdVM.MustVm().MustName(), createdVM.MustVm().MustId())
+	}
+	// extend the disk if requested size is bigger then template. We won't support shrinking it.
```
Comment (bennyz) [inline pkg/cloud/ovirt/clients/machineservice.go:218]: than

**Pull request merged**